apps/ledger-live-desktop/src/renderer/hooks/swap-migrations/useSwapLiveAppHook.tsx

```diff
@@ -51,16 +59,20 @@ export const useSwapLiveAppHook = (props: UseSwapLiveAppHookProps) => {
         loading = swapTransaction.bridgePending || exchangeRatesState.status === "loading";
       }
 
-      updateSwapWebProps({
+      const newSwapWebProps = {
         provider,
         ...getExchangeSDKParams(),
         fromParentAccountId,
-        cacheKey: v4(),
         error: !!swapError,
         loading,
         providerRedirectURL,
         swapApiBase: SWAP_API_BASE,
-      });
+      };
+
+      if (!isEqual(newSwapWebProps, swapWebProps)) {
+        setSwapWebProps(newSwapWebProps);
```

Comment:
setting state inside a useEffect that is watching that state could create an infinite loop of re-rendering.

The new `swapWebProps` state in this hook seems to mirror whatever the state is that is being updated by `updateSwapWebProps`. So do we not have access to that SwapWebProps state?

I wonder if we could store SwapWebProps in a ref, if we must do a comparison? But it strikes me that we should only need to watch for the `cacheKey` changing. Or update in a more selective reducer, piece by piece, instead of lumping all SwapWebProps into a single object?

Reply:
The cacheKey updates on every render that's what war creating the problem since `updateSwapWebProps` is also a setter for a parent component
I can put `swapWebProps` in a ref and compare them in the same way